recipes-support/provisioning/files/test-provision-tpm.sh

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env bash
+#===============================================================================
+#
+#  provision-tpm.sh
+#
+#  Copyright (C) 2021 by PHYTEC Messtechnik GmbH
+#  All rights reserved.
+#
+#  This program is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU General Public License version 2 as published by
+#  the Free Software Foundation.
+#
+#===============================================================================
+
+printf "======================================\n" > /dev/console
+printf "PHYTEC TPM2 Test Provision Version 0.2\n" > /dev/console
+printf "======================================\n" > /dev/console
+
+# Get CPU UID
+get_cpu_uid() {
+    val=$(cat /sys/devices/soc0/soc_uid)
+    echo "${val}"
+}
+
+set -e
+trap end EXIT
+end() {
+    if [ "$?" -ne 0 ]; then
+        printf "PHYTEC-TPM2-Provision:fail! \n" > /dev/console
+    fi
+}
+
+# Variable declaration
+CA_DIR=`mktemp -d -t tpm2tmpca.XXXXXX`
+export CA_DIR
+
+tpm2pkcs11tool='pkcs11-tool --module /usr/lib/pkcs11/libtpm2_pkcs11.so'
+# Generate SO-PIN
+TPM_SOPIN=$(tpm2_getrandom --hex 8 | tr -dc 0-9)
+TPM_PIN=$(get_cpu_uid | head -c${1:-7})
+
+# With Testing Intermediate CA from PKCS#12 container
+setup_casubtest() {
+    CA_P12="${PWD}/Testing_Sub_CA.p12"
+    export CA_PEM="${CA_DIR}/subca.pem"
+    export CA_KEY="${CA_DIR}/subca.key"
+    # Extract CA
+    openssl pkcs12 -in "${CA_P12}" -out "${CA_PEM}" -clcerts -nokeys
+    # Extract Keys
+    openssl pkcs12 -in "${CA_P12}" -out "${CA_KEY}" -nocerts -nodes
```

Comment:
The private key of the ca is on the device. An attacker could simply copy it and create his own certificate. I assume that we just create the CSR on the device and sign it using the Phytec Smartcard HSM in the production phase, right?

Reply:
yes this is only for the first test or integration tests. For the production we use the Smartcard HSM, which will integrated in the  next step.